bugs/lp-1850988.txt

```diff
@@ -4,32 +4,63 @@ https://bugs.launchpad.net/ubuntu/+source/cloud-init/+bug/1850988
 === Begin SRU Template ===
 [Impact]
 
+Users passing a network configuration that includes a "static6" type subnet
+would not have that subnet configured correctly.
+
 [Test Case]
 
-[Regression Potential]
+This test function will launch a lxd container with an affected configuration,
+and check that the static6 interface is configured as expected.
 
-[Other Info]
-Upstream commit at
-  https://git.launchpad.net/cloud-init/commit/?id=XXXXXXXXXX
+(Note that it relies on manual inspection of the output for validation, and
+because the interfaces being configured don't actually have network
+connectivity, the "wait" step takes longer than usual.)
+
+```
+lp1850988() {
+    release=$1
+    ref=$release-proposed
+    name=$release-lp1850988
 
-cat > some.yaml <<EOF
-#cloud-config
-hostname: sru-test
+    mac_address="00:50:56:89:b7:48"
+    cat > network.yaml << EOF
+version: 1
+config:
+    - subnets:
+        - control: auto
+          netmask: 255.255.255.0
+          type: static
+          address: 1.1.1.4
+        - netmask: 64
+          type: static6
+          address: 2600::10
+      type: physical
+      name: ens192
+      mac_address: $mac_address
 EOF
 
-# Perform individual bug tests using lxd for speed where possible
-for series in xenial bionic cosmic disco; do
-   echo '=== BEGIN ' $series
-   ref=$series-proposed
-   lxc delete test-$series --force 2> /dev/null
-   lxc init  test-$series
-   lxc config set test-$series user.user-data - < some.yaml
-   lxc start test-$series
-   lxc exec test-$series -- cloud-init status --wait --long
+    # https://github.com/cloud-init/qa-scripts/blob/master/scripts/lxc-proposed-snapshot
+    lxc-proposed-snapshot --proposed --publish $release $ref
+    lxc launch -c user.network-config="$(cat network.yaml)" -c volatile.eth0.hwaddr=$mac_address $ref $name
+
+    lxc exec $name -- cloud-init status --wait
 
-   # Add specific tests
-   echo '=== END ' $series
-done
+    echo -n "This should be 1: "
+    lxc exec $name -- ip addr | grep -c 'inet6 2600::10/64'
```

Comment:
Minimally we should probably also be doing a grep for 
Tracebacks in cloud-init.log as well.

Reply:
The tracebacks only occur on CentOS; on Ubuntu we silently fail to configure the IPv6 subnet.